energy-integration/fees-collector/src/redistribute_rewards.rs

```diff
@@ -0,0 +1,75 @@
+use common_types::{PaymentsVec, TokenAmountPair, Week};
+use weekly_rewards_splitting::USER_MAX_CLAIM_WEEKS;
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait RedistributeRewardsModule:
+    crate::fees_accumulation::FeesAccumulationModule
+    + crate::config::ConfigModule
+    + crate::events::FeesCollectorEventsModule
+    + week_timekeeping::WeekTimekeepingModule
+    + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[only_admin]
+    #[endpoint(redistributeRewards)]
+    fn redistribute_rewards(&self, start_week: Week, end_week: Week) {
+        let collect_rewards_offset = USER_MAX_CLAIM_WEEKS + 1;
+        let current_week = self.get_current_week();
+        require!(
+            current_week > collect_rewards_offset,
+            "Current week must be higher than the week offset"
+        );
+        require!(start_week <= end_week, "Invalid week numbers");
+        require!(
+            end_week <= current_week - collect_rewards_offset,
+            "Invalid end week"
+        );
+
+        let all_tokens = self.all_tokens().get();
+        let mut all_rewards = ManagedVec::new();
+        for token_id in &all_tokens {
+            all_rewards.push(TokenAmountPair::new(token_id, BigUint::zero()));
+        }
+
+        for week in start_week..=end_week {
+            self.accumulate_remaining_rewards_single_week(&mut all_rewards, &all_tokens, week);
+        }
+
+        for reward_entry in &all_rewards {
+            if reward_entry.amount == 0 {
+                continue;
+            }
+
+            self.accumulated_fees(current_week, &reward_entry.token)
```

Comment:
So we just accumulate the remaining rewards, with no means of claiming them?

Reply:
No, they're put in current week storage, so users can claim them starting next week.